mlflow/utils/environment.py

```diff
@@ -16,10 +16,69 @@
 """
 
 _CONDA_ENV_FILE_NAME = "conda.yaml"
+_PYTHON_ENV_FILE_NAME = "python-env.yaml"
 _REQUIREMENTS_FILE_NAME = "requirements.txt"
 _CONSTRAINTS_FILE_NAME = "constraints.txt"
 
 
+class _PythonEnv:
```

Comment:
```suggestion
class PythonEnv:
```

Maybe this class should be public to allow users to construct a python env programatically.


```python
env = PythonEnv(...)
mlflow.*.log_model(.., python_env=env)
```